core/base/v7/inc/ROOT/TDirectoryEntry.hxx

```diff
@@ -33,7 +31,7 @@ public:
 
 private:
   time_point_t fDate = clock_t::now(); ///< Time of last change
-  TClass* fType;
+  const std::type_info* fType;
```

Comment:
I've thought that the main goal of searching for alternative to TClass-based cast is to completely get rid of TClass in TDirectory(Entry).
Is there any reason not to use TClass::DynamicCast keepeng TClass around then?